pkg/controller/replicaset/replica_set.go

```diff
@@ -235,10 +235,8 @@ func (rsc *ReplicaSetController) getReplicaSetsWithSameController(rs *apps.Repli
 		relatedRSs = append(relatedRSs, obj.(*apps.ReplicaSet))
 	}
 
-	// The if check is used to avoid the overhead for the KObjs call, see
-	// https://github.com/kubernetes/kubernetes/issues/106945.
 	if klogV := klog.V(2); klogV.Enabled() {
-		klogV.InfoS("Found related ReplicaSets", "replicaSet", klog.KObj(rs), "relatedReplicaSets", klog.KObjs(relatedRSs))
+		klogV.InfoS("Found related ReplicaSets", "replicaSet", klog.KObj(rs), "relatedReplicaSets", klog.KObjSlice(relatedRSs))
```

Comment:
Here you can simplify the code and remove the if check.

Reply:
I wonder whether this is related to KObjSlice and/or KObj at all. In other words, what is the performance difference between

```
klog.V(2).InfoS("Some key/value pairs", "a", 1, "b", "x")
```

and

```
if klogV := klog.V(2); klogV.Enabled() {
   klogV.InfoS("Some key/value pairs", "a", 1, "b", "x")
}
```

It is clear that we pay some price for using the simpler one-line version, but how high is that price? I don't know.

Can you perhaps investigate? That'll put the slowdown for passing KObj and KObjSlice into perspective. If they are still "more expensive", then keeping the if check makes sense. If they are equally expensive, then it probably doesn't.

Reply:
/ack . Let me do that and update back.

Reply:
I had a look and [came to the conclusion](https://kubernetes.slack.com/archives/CG3518SFJ/p1666892411521019?thread_ts=1662480510.933609&cid=CG3518SFJ) that the `if klogV := klog.V(2); klogV.Enabled() ` variant is always faster. It matters a little bit more for `KObjSlice` than normal scalars because it needs additional memory allocations.

So lets keep the if check here.